UI/js/signup.js

```diff
@@ -0,0 +1,267 @@
+const firstName = document.getElementById('register-firstname'),
+  firstNameErrorMessage = document.getElementById('register-firstname-error'),
+  lastName = document.getElementById('register-lastname'),
+  lastNameErrorMessage = document.getElementById('register-lastname-error'),
+  email = document.getElementById('register-email'),
+  emailErrorMessage = document.getElementById('register-email-error'),
+  password = document.getElementById('register-password'),
+  passwordErrorMessage = document.getElementById('register-password-error'),
+  confirmPassword = document.getElementById('register-confirm-password'),
+  confirmPasswordErrorMessage = document.getElementById('register-confirm-password-error'),
+  signupButton = document.querySelector('#signup-form-btn');
+
+// regex references
+// https://stackoverflow.com/questions/20003870/how-do-i-create-regex-of-min-8-max-16-alphabetic-numbers-and-no-space
+// https://stackoverflow.com/questions/201323/how-to-validate-an-email-address-using-a-regular-expression
+// https://stackoverflow.com/questions/9289451/regular-expression-for-alphabets-with-spaces
+
+const errors = {};
+
+signupButton.addEventListener('mouseenter', () => {
+  if (!firstName.value.trim()) {
+    errors.firstName = 'First name is required';
```

Comment:
It seems like these functions and the next do almost the same thing. I see a lot of code repetition. Can't we abstract the checks for lastName and firstName into a function parameterized by which field you want to validate?